File: 331-verify-preorder-serialization-of-a-binary-tree/verify-preorder-serialization-of-a-binary-tree.py
class Solution:
    def isValidSerialization(self, preorder: str) -> bool:
        degree=1
        for ch in preorder.split(','):
            degree=degree-1
            if degree<0:
                return False
            if ch!='#':
                degree=degree+2
        return degree==0
            

File: 331-verify-preorder-serialization-of-a-binary-tree/verify-preorder-serialization-of-a-binary-tree.java
class Solution {
    public boolean isValidSerialization(String preorder) {
        String[] list=preorder.split(",");
        int degree=1;
        for(String ch: list){
            degree--;
            if(degree<0){
                return false;
            }

            if(!ch.equals("#")){
                degree=degree+2;
            }
        }
        return degree==0;
    }
}